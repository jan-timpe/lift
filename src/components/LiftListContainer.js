import React, { Component } from 'react'
import { connect } from 'react-redux'
import { addLift } from '../actions'
import LiftForm from './LiftForm'
import LiftList from './LiftList'


const mapStateToProps = (state) => ({
  lifts: state.lifts
})

const mapDispatchToProps = (dispatch) => ({
  addLift: args => dispatch(addLift(args))
})


class LiftListContainer extends Component {

  render() {
    return (
      <div className="LiftListContainer">
        <LiftForm addLift={this.props.addLift} />
        <LiftList lifts={this.props.lifts} />
      </div>
    )
  }


}

export default connect(mapStateToProps, mapDispatchToProps)(LiftListContainer);