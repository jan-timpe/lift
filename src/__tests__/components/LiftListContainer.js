import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import configureStore from 'redux-mock-store'
import LiftListContainer from '../../components/LiftListContainer'


const mockStore = configureStore()

const initialState = [{
  lifts: [{
    id: 0,
    name: 'Squat'
  }, {
    id: 1,
    name: 'Bench'
  }],
  liftResults: [{
    id: 0,
    liftId: 0,
    weight: 205,
    date: '7/1/2018'
  }]
}]

const store = mockStore(initialState)


describe(`LiftListContainer`, () => {

  it(`should render`, () => {
    const wrapper = shallow(<LiftListContainer store={store} />)
    const component = wrapper.dive()
    expect(toJson(component)).toMatchSnapshot()
  })

})