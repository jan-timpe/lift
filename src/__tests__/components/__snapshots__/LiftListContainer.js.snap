// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiftListContainer should render 1`] = `
<div
  className="LiftListContainer"
>
  <LiftForm
    addLift={[Function]}
  />
  <LiftList />
</div>
`;
